projects/storefrontlib/src/cms-components/myaccount/order/replenishment-order-history/replenishment-order-history.component.spec.ts

```diff
@@ -0,0 +1,221 @@
+};
+
+const replenishmentOrderHistroy = new BehaviorSubject(mockReplenishmentOrders);
+
+// mock the service
+//
+
+@Component({
+  template: '',
+  selector: 'cx-pagination',
+})
+class MockPaginationComponent {
+  @Input() pagination;
+  @Output() viewPageEvent = new EventEmitter<string>();
+}
+@Component({
+  template: '',
+  selector: 'cx-sorting',
+})
+class MockSortingComponent {
+  @Input() sortOptions;
+  @Input() sortLabels;
+  @Input() selectedOption;
+  @Input() placeholder;
+  @Output() sortListEvent = new EventEmitter<string>();
+}
+
+class MockReplenishmentOrderCancellationLaunchDialogService {
+  openDialog(_openElement?: ElementRef, _vcr?: ViewContainerRef) {}
+}
+
+@Pipe({
+  name: 'cxUrl',
+})
+class MockUrlPipe implements PipeTransform {
+  transform() {}
+}
+
+class MockUserOrderService {
+  getReplenishmentOrderHistoryList(): Observable<ReplenishmentOrderList> {
+    return replenishmentOrderHistroy.asObservable();
+  }
+  getReplenishmentOrderHistoryListSuccess(): Observable<boolean> {
+    return of(true);
+  }
+  loadReplenishmentOrderList(
+    _userId: string,
+    _pageSize: number,
+    _currentPage?: number,
+    _sort?: string
+  ): void {}
+  clearReplenishmentOrderList() {}
+}
+
+class MockRoutingService {
+  go() {}
+}
+
+describe('ReplenishmentOrderHistoryComponent', () => {
+  let component: ReplenishmentOrderHistoryComponent;
+  let fixture: ComponentFixture<ReplenishmentOrderHistoryComponent>;
+  let userService: UserReplenishmentOrderService;
+  let routingService: RoutingService;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [RouterTestingModule, I18nTestingModule],
+      declarations: [
+        ReplenishmentOrderHistoryComponent,
+        MockUrlPipe,
+        MockPaginationComponent,
+        MockSortingComponent,
+      ],
+      providers: [
+        { provide: RoutingService, useClass: MockRoutingService },
+        {
+          provide: UserReplenishmentOrderService,
+          useClass: MockUserOrderService,
+        },
+        {
+          provide: ReplenishmentOrderCancellationLaunchDialogService,
+          useClass: MockReplenishmentOrderCancellationLaunchDialogService,
+        },
+      ],
+    }).compileComponents();
+
+    userService = TestBed.inject(UserReplenishmentOrderService);
+    routingService = TestBed.inject(RoutingService);
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ReplenishmentOrderHistoryComponent);
+    component = fixture.componentInstance;
+  });
+
+  it('should be created', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should read replenishment order list', () => {
+    let replenishmentOrders: ReplenishmentOrderList;
+    component.replenishmentOrders$
+      .subscribe((value) => {
+        replenishmentOrders = value;
+      })
+      .unsubscribe();
+
+    expect(replenishmentOrders).toEqual(mockReplenishmentOrders);
+  });
+
+  it('should redirect when clicking on order id', () => {
```

Comment:
replenishment order code and not order id

when we navigate to a specific replenishment code, we take the data not from `code`, but from `ReplenishmentOrderCode`. 

change to replenishment order id or replenishment order code to be more descriptive. Similar tor other `it` function within this PR